fitbenchmarking/results_processing/performance_profiler.py

```diff
@@ -171,27 +175,20 @@ def create_plot(ax, step_values, keys):
 
     no_failures = np.zeros(len(step_values), dtype=np.int8)
     huge = 1.0e20  # set a large value as a proxy for infinity
-    for i, solver_values in enumerate(step_values):
+    plot_points = np.linspace(0.0, 1.0, step_values[0].size)
+    plot_points = np.append(plot_points, 1.0)
+    
+    for i, (solver, solver_values) in enumerate(zip(labels, step_values)):
         inf_indices = np.where(solver_values > huge)
         solver_values[inf_indices] = huge
-        if inf_indices:
+        if inf_indices[0].size > 0:
             no_failures[i] = len(inf_indices[0])
-
-    labels = [key for key in keys]
-    for i, solver in enumerate(labels):
-        if no_failures[i]:
             labels[i] = "{} ({} failures)".format(solver, no_failures[i])
-
-    plot_points = np.linspace(0.0, 1.0, step_values[0].size)
-    for s, step_value in enumerate(step_values):
-        step_value = np.append(step_value,huge)
-        if s == 0:
-            plot_points = np.append(plot_points,1.0)
-        ax.step(step_value,
+        solver_values = np.append(solver_values,huge)
+        ax.step(solver_values,
                 plot_points,
-                label=labels[s],
-                linestyle=lines[(s % len(lines))],
-                color=colors[(s % len(colors))],
+                label=labels[i],
```

Comment:
```suggestion
                label=label,
```

Reply:
I'm amending the labels array that comes in from `acc.keys()` here, so that we don't make a copy if we already have the information.  If I use this suggestion, I'd have to set `label=labels[i]` at the top of the loop.  I don't see the benefit of that at the moment -- let me know if I'm missing something.

Reply:
fixed, as discussed